crates/bevy_ecs/src/query/mod.rs

```diff
@@ -657,6 +661,115 @@ count():       {count}"#
         }
     }
 
+    #[test]
+    fn batched_queries() {
+        let mut world = World::new();
+
+        world.spawn_batch(
+            (0..127)
+                .into_iter()
+                .map(|i| (A(4 * i), B(4 * i + 1), C(4 * i + 2), D(4 * i + 3))),
+        );
+
+        fn system_compute(mut q: Query<(&mut A, &B, &C, &D)>) {
+            let mut scalar_counter = 0;
+            let mut batch_counter = 0;
+
+            q.for_each_mut_batched::<4, 16, _, _>(
```

Comment:
Needs a comment describing how these const generic values are set.

Reply:
Done!  More will be coming on this front too.  As it turns out, it was possible to use `impl` syntax to avoid having to pass in the `_` arguments for the lambdas.